rust/template/differential_datalog/src/record/mod.rs

```diff
@@ -826,6 +831,170 @@ impl<T: FromRecord + Ord> Mutator<BTreeSet<T>> for Record {
     }
 }
 
+impl<T> FromRecord for Arc<T>
```

Comment:
I was surprised to see these impls given that we already have all the traits implemented for `Ref<>` in `ddlog_std.rs`, but I guess there's no harm in having them: if someone wants to construct extern types using `Arc` and friends, they will be able to take advantage of these existing impls.